app/javascript/components/student/editor/EditorStatusSummary.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { Action, EditorStatus } from '../Editor'
+
+export function EditorStatusSummary({
+  status,
+  dispatch,
+}: {
+  status?: EditorStatus
+  dispatch: (action: Action) => void
+}) {
+  switch (status) {
```

Comment:
I assume you're expecting more statuses to handle besided the one that is now being used? If not, this could also be an if-statement.